# เนื้อหาโดยรวม
* ตัวแปร (Variables) คืออะไร
* การตั้งชื่อตัวแปรที่ถูกต้อง
* ชนิดข้อมูลพื้นฐาน (String, Integer, Float, Boolean)
* การตรวจสอบชนิดข้อมูล
* การแปลงชนิดข้อมูล

## 1. ตัวแปร (Variables) คืออะไร?

ตัวแปรเปรียบเสมือนกล่อง<b>ที่ใช้เก็บข้อมูล</b> โดยเราสามารถ<b>นำข้อมูลมาเก็บไว้และนำไปใช้ได้ในภายหลัง</b>

การสร้างตัวแปรใน Python ทำได้โดยประกาศ `ชื่อตัวแปร = ค่า`

In [ ]:
# ตัวอย่างการประกาศตัวแปร
name = 'Louis'     # string
surename = "MakerLab" # string
age = 25           # interger
height = 175.0     # float
is_student = False  # boolean

# การแสดงค่าของตัวแปร
print("ชื่อ:", name)
print("นามสกุล: ", surename)
print("อายุ:", age)
print("ส่วนสูง:", height, "เซนติเมตร")
print("เป็นนักเรียน:", is_student)

ชื่อ: Louis
นามสกุล:  MakerLab
อายุ: 25
ส่วนสูง: 175.0 เซนติเมตร
เป็นนักเรียน: False


## 2. การตั้งชื่อตัวแปรที่ถูกต้อง

กฎการตั้งชื่อตัวแปรใน Python:
* ต้องขึ้นต้นด้วยตัวอักษรหรือเครื่องหมาย underscore (_)
* ประกอบด้วยตัวอักษร ตัวเลข หรือ underscore
* ตัวพิมพ์เล็กและตัวพิมพ์ใหญ่ถือว่าเป็นคนละตัวกัน (case-sensitive)
* ห้ามใช้คำสงวน (reserved keywords) ของ Python เช่น if, else, while, for เป็นชื่อตัวแปร

https://khalilstemmler.com/blogs/camel-case-snake-case-pascal-case/


In [ ]:
# ตัวอย่างการตั้งชื่อตัวแปรที่ถูกต้อง
user_name = "John"    # แนะนำให้ใช้ snake_case สำหรับชื่อตัวแปร
age1 = 30             # ใช้ตัวเลขตามหลังได้
_private = "secret"   # เริ่มด้วย underscore ได้

1
2


## 3. ชนิดข้อมูลพื้นฐาน (Basic Data Types)

Python มีชนิดข้อมูลพื้นฐาน 4 ประเภท:
1. String (str): ข้อความ
2. Integer (int): จำนวนเต็ม
3. Float (float): จำนวนทศนิยม
4. Boolean (bool): ค่าความจริง (True/False)

### 3.1 String - ข้อความ

In [41]:
# String คือข้อความที่อยู่ในเครื่องหมาย quote (' หรือ ")
first_name = "Alice"
last_name = 'Smith'
address = "123 ถนนสุขุมวิท กรุงเทพฯ"

# การจัดการกับ String
# 1. การต่อข้อความ (Concatenation)
full_name = first_name + " " + last_name
print(full_name)
# 2. การทำซ้ำ (Repetition)
times = 100
display_text = "*" * times
print(display_text)
# 3. การเข้าถึงตัวอักษรในข้อความ (Indexing)
print(first_name[4])
print(first_name[-5])
# 4. การตัดข้อความ (Slicing)
print("สามตัวแรกของที่อยู่: ", address[0:3])
# 5. ความยาวของ String
print(len(first_name))
print(first_name[len(first_name)-1])
# 6. Method ของ String
print("Upper Case: ", first_name.upper())
print("Lower Case: ", first_name.lower())
input = " Louis "
print(input.strip())

Alice Smith
****************************************************************************************************
e
A
สามตัวแรกของที่อยู่:  123
5
e
Upper Case:  ALICE
Lower Case:  alice
Louis


### 3.2 Integer - จำนวนเต็ม

In [ ]:
# Integer คือจำนวนเต็ม (ไม่มีจุดทศนิยม)
age = 25
temperature = -10
count = 1_000_000  # สามารถใช้ underscore เพื่อให้อ่านง่ายขึ้น

a = 15
b = 8

print(a+b) 
print(a-b)
print(a*b)
print(a/b)
print(a//b)
print(a%b)
print(a**b)

23
7
120
1.875
1
7
2562890625


### 3.3 Float - จำนวนทศนิยม

In [60]:
# Float คือจำนวนที่มีจุดทศนิยม
height = 165.5
weight = 55.8
pi = 3.14159
science_notation = 6.022e-1  # ใช้สัญกรณ์ทางวิทยาศาสตร์ได้

print("ค่า pi:", pi)
print("จำนวนในรูปแบบสัญกรณ์วิทยาศาสตร์:", science_notation)

sum = 0.1 + 0.25
print(round(sum, 2))

ค่า pi: 3.14159
จำนวนในรูปแบบสัญกรณ์วิทยาศาสตร์: 0.6022
0.35


### 3.4 Boolean - ค่าความจริง

In [ ]:
# Boolean มีแค่ 2 ค่า: True และ False
is_student = True
is_working = False

print("เป็นนักเรียน?", is_student)
print("กำลังทำงาน?", is_working)
# การใช้ Boolean ในการเปรียบเทียบ
age = 20
is_adult = age>=20
print("Adult ?: ",is_adult) 
# การใช้ Boolean กับตรรกะ (Logical operators and, or, not)
is_student = True
has_student_id = False
print("สามารถเข้าห้องสมุดได้มั้ย: ", is_student and has_student_id)
# ค่าที่ถือว่าเป็น False ใน Python (0, "", '', [], None)
input = None
if input:
    print("True: ",input)
else:
    print("False: ",input)
# ค่าที่ถือว่าเป็น True ใน Python
print("bool(1) =", bool(1))        # จำนวนที่ไม่ใช่ศูนย์ถือว่าเป็น True
print("bool('hello') =", bool("hello")) # string ที่ไม่ว่างถือว่าเป็น True
print("bool([1, 2]) =", bool([1, 2]))  # list ที่ไม่ว่างถือว่าเป็น True

เป็นนักเรียน? True
กำลังทำงาน? False
Adult ?:  True
สามารถเข้าห้องสมุดได้มั้ย:  False
False:  None


## 4. การตรวจสอบชนิดข้อมูล (Type Checking)

In [103]:
# การใช้ฟังก์ชัน type() เพื่อตรวจสอบชนิดข้อมูล
name = "Alice"
age = 25
height = 165.5
is_student = True

input = None
print(type(input))

input = 1
print(type(input))

input = 10.2
print(type(input))

input = "Text"
print(type(input))

# การเปรียบเทียบชนิดข้อมูล (str, int, float)
print("input เป็น str หรือไม่: ", type(input) is str)
input = 5
print("input เป็น int หรือไม่: ", type(input) is int)
input = None
print("input เป็น None หรือไม่: ", input is None)

# การใช้ isinstance() เพื่อตรวจสอบชนิดข้อมูล
input = "text"
print("input เป็น int หรือไม่: ", isinstance(input, int))
print("input เป็น str หรือไม่: ", isinstance(input, str))

<class 'NoneType'>
<class 'int'>
<class 'float'>
<class 'str'>
input เป็น str หรือไม่:  True
input เป็น int หรือไม่:  True
input เป็น None หรือไม่:  True
input เป็น int หรือไม่:  False
input เป็น str หรือไม่:  True


## 5. การแปลงชนิดข้อมูล (Type Conversion)

In [ ]:
# การแปลงชนิดข้อมูล (Type Conversion)
age = 25
age = str(age)
print(type(age))

age = "100"
age = int(age)
print(type(age))

height = "150.1"
height = float(height)
print(type(height))


# 0, '', []
print(bool(1))
print(bool('as'))
print(bool([1,2]))

number = round(float("10.5"), 0)
number = int(number)
print(type(number))

try:
    input = "Alice"
    age = int(input)
except ValueError as error: 
    input = 0
    age = int(input)
    print(type(age))
    print(error)

<class 'str'>
<class 'int'>
<class 'float'>
True
True
True
<class 'int'>
<class 'int'>
invalid literal for int() with base 10: 'Alice'


## 6. แบบฝึกหัด (Exercises)

In [ ]:
# แบบฝึกหัดที่ 1: สร้างตัวแปรใหม่สำหรับเก็บข้อมูลส่วนตัว
# TODO: สร้างตัวแปรสำหรับ ชื่อเล่น, ปีเกิด, น้ำหนัก และสถานะการแต่งงาน

# แบบฝึกหัดที่ 2: คำนวณอายุจากปีเกิด
# TODO: สร้างตัวแปรเก็บปีปัจจุบัน และคำนวณอายุจากปีเกิดที่กำหนด

# แบบฝึกหัดที่ 3: สร้าง profile แนะนำตัวจากข้อมูลส่วนตัว
# TODO: นำข้อมูลทั้งหมดมาสร้างเป็นประโยคแนะนำตัว

# แบบฝึกหัดที่ 4: ตรวจสอบเงื่อนไข
# TODO: ตรวจสอบว่าอายุมากกว่า 18 ปีหรือไม่ และผลลัพธ์เป็น Boolean

# แบบฝึกหัดที่ 5: แปลงชนิดข้อมูล
# TODO: รับข้อมูลอายุเป็น string และแปลงเป็น integer

ชื่อ 123, อายุ 25


## 7. สรุป

ในบทเรียนนี้เราได้เรียนรู้เกี่ยวกับ:

1. **ตัวแปร (Variables)** - พื้นที่เก็บข้อมูลที่สามารถเปลี่ยนแปลงได้
2. **การตั้งชื่อตัวแปร** - กฎและวิธีการตั้งชื่อตัวแปรที่ดี
3. **ชนิดข้อมูลพื้นฐาน**
   - **String (str)** - ข้อความ เช่น "Hello"
   - **Integer (int)** - จำนวนเต็ม เช่น 42
   - **Float (float)** - จำนวนทศนิยม เช่น 3.14
   - **Boolean (bool)** - ค่าความจริง True/False
4. **การตรวจสอบชนิดข้อมูล** - ใช้ฟังก์ชัน type() และ isinstance()
5. **การแปลงชนิดข้อมูล** - ใช้ฟังก์ชัน str(), int(), float(), bool()

สิ่งสำคัญคือการเลือกใช้ชนิดข้อมูลที่เหมาะสมกับงานและการจัดการกับข้อผิดพลาดที่อาจเกิดขึ้นจากการแปลงชนิดข้อมูล

## 8. เฉลยแบบฝึกหัด

In [ ]:
# เฉลยแบบฝึกหัดที่ 1
nickname = "โจ้"
birth_year = 2000
weight = 65.7
is_married = False

print("ชื่อเล่น:", nickname)
print("ปีเกิด:", birth_year)
print("น้ำหนัก:", weight)
print("สถานะการแต่งงาน:", is_married)
print("ชนิดข้อมูล:", type(nickname), type(birth_year), type(weight), type(is_married))

# เฉลยแบบฝึกหัดที่ 2
current_year = 2025
age = current_year - birth_year
print(f"อายุ: {age} ปี")

# เฉลยแบบฝึกหัดที่ 3
status = "แต่งงาน" if is_married else "โสด"
profile = f"สวัสดีครับ ผมชื่อ {nickname} อายุ {age} ปี น้ำหนัก {weight} กก. สถานะ: {status}"
print(profile)

# เฉลยแบบฝึกหัดที่ 4
is_adult = age >= 18
print(f"อายุ {age} ปีถือว่าเป็นผู้ใหญ่หรือไม่? {is_adult}")

# เฉลยแบบฝึกหัดที่ 5
age_input = "25"  # สมมติว่านี่คือข้อมูลที่รับมา
age_value = int(age_input)
print(f"อายุก่อนแปลง: {age_input} (ชนิด: {type(age_input)})")
print(f"อายุหลังแปลง: {age_value} (ชนิด: {type(age_value)})")
print(f"ปีที่เกิด: {current_year - age_value}")